### ISSUE TITLE
MD051/link-fragments false positive when code syntax is used

### ISSUE THREAD

**Issue by undergroundwires** [DavidAnson/markdownlint#591]
First of all thank you for the great tool. I use it on almost all of my projects. After upgrading to new version, I realized that MD051 fails to detect valid headers when code ``` backticks are used.

Example:

```md

## `Header`

[Link](#Header) to header.
```

This is valid link and valid markdown but I get `MD051/link-fragments Link fragments should be valid` error.

Here is the [markdown](https://github.com/undergroundwires/privacy.sexy/blob/bd9f4bcaf8a4319473ae5ca2d25f0470276d23d8/docs/collection-files.md) I use it on.

**Comment by nschonni** [comment]
It complains about the anchor not being lowercase, not the backticks

**Comment by undergroundwires** [comment]
Thank you for super fast response @nschonni , I'm too stupid, sorry for taking your time.

**Comment by nschonni** [comment]
No worries, it's one I hit before too. Not sure if the messaging needs to be updated

**Comment by DavidAnson** [comment]
I made myself a note to look into that.

**Comment by MFTabriz** [comment]
Does the anchor really need to be in lowercase? RfC 3986 does not define the URI fragment as case in-sensitive.

**Comment by DavidAnson** [comment]
That's correct, GitHub appears to always use lowercase, and so that is the form this rule requires.
